### DISCUSSION TITLE
Welcome to Toit language discussions!

### DISCUSSION THREAD

**Discussion by kasperl** [toitlang/toit#2] [Announcements]
## Welcome! 👋

The Toit language implementation is free and open source and we believe it can become a great addition to the developer toolbox of everyone looking for a robust and high-level foundation for their microcontroller programming.

We’re using Discussions as a place to connect with other members of our community. We hope that you:

* Ask questions you’re wondering about.
* Share ideas.
* Engage with other community members.
* Welcome others and are open-minded. Remember that this is a community we build together 💪.

To get started, feel free to comment below with an introduction of yourself and tell us about your experience building for the ESP32 family of processors.

**Comment by kasperl** [comment]
I'll start! 🎉

I'm Kasper Lund, co-founder of [Toit.io](https://toit.io/) and I spent a couple of decades working on programming language implementations. I just can't seem to get enough of those.

I was a founding engineer on [Google's V8 JavaScript engine](https://v8.dev/), the tech lead on the first [adaptively optimizing compiler for JavaScript](https://blog.chromium.org/2010/12/new-crankshaft-for-v8.html), and I co-led the design and implementation of the [Dart language](https://dart.dev/) used by [Flutter](https://flutter.dev/).

I'm blown away with what you can do with an ESP32 and I can't wait to see what the future brings for the new breed of microcontrollers.

**Comment by floitsch** [comment]
I will go second.

Hi, I'm Florian Loitsch, also a co-founder of Toit.io. For the last 10 years I have worked with Kasper, so lots of similarities there: some VM work, and then Dart. Specifically I was responsible for the language and libraries, and the dart2js compiler (though never at the same time).

I intend to make the open-source version of Toit as useful as possible. If you are missing features, or see other possible improvements let me know.

**Comment by lask** [comment]
Hi, my name is Lau. I have been an employee at Toit for about two years. Before I started at Toit, I did a PhD at Aarhus University about proving high-level language properties guaranteed by capability machines. We will probably not see capability machines with proven high-level guarantees for quite some time. We will, however, see millions of ESP32s, so I joined Toit to make a difference today.
